core/src/main/kotlin/fr/sncf/osrd/api/api_v2/path_properties/PathPropResultConverter.kt

```diff
@@ -0,0 +1,120 @@
+package fr.sncf.osrd.api.api_v2.path_properties
+
+import fr.sncf.osrd.api.pathfinding.toRJSLineString
+import fr.sncf.osrd.railjson.schema.geom.RJSLineString
+import fr.sncf.osrd.sim_infra.api.PathProperties
+import fr.sncf.osrd.utils.DistanceRangeMap
+import fr.sncf.osrd.utils.DistanceRangeMapImpl
+import fr.sncf.osrd.utils.units.Distance
+
+fun makePathPropResult(pathProperties: PathProperties, rsSupportedElectrifications: List<String>): PathPropResult {
+    return PathPropResult(
+        makeSlopes(pathProperties),
+        makeGradients(pathProperties),
+        makeElectrifications(pathProperties, rsSupportedElectrifications),
+        makeGeographic(pathProperties)
+    )
+}
+
+private fun makeSlopes(pathProperties: PathProperties): RangeValues<Double> {
+    return makeRangeValues(pathProperties.getSlopes())
+}
+
+private fun makeGradients(pathProperties: PathProperties): RangeValues<Double> {
+    return makeRangeValues(pathProperties.getGradients())
+}
+
+private fun makeElectrifications(pathProperties: PathProperties, rsSupportedElectrifications: List<String>): RangeValues<Electrification> {
+    // If there are no supported electrifications, return NonElectrified along the whole path
+    if (rsSupportedElectrifications.isEmpty())
+        return RangeValues(listOf(Distance.ZERO, pathProperties.getLength().distance), listOf(Electrification.NonElectrified))
+    val electrifications = makeElectrificationMap(pathProperties.getElectrification(), rsSupportedElectrifications)
+    val neutralSections = makeElectrificationMap(pathProperties.getNeutralSections())
+    val combinedEntries = (electrifications + neutralSections).sortedBy { it.lower }
+    val mergedEntries = mutableListOf<DistanceRangeMap.RangeMapEntry<Electrification>>()
+    var i = 0
+    var currentLower = Distance.ZERO
+    while (i < combinedEntries.size) {
```

Comment:
With that much complexity, we could use `com.google.common.collect.RangeMap`. It has a lot more utility functions builtin, in particular methods to merge different maps with easy ways to handle overlaps. We could convert to `RangeMap`, do all that processing, and then convert back to a `RangeValues`. It would be a little less efficient, but I'd be more confident in the reliability of that code